### DISCUSSION TITLE
1C charging rate fills battery in 20 min

### DISCUSSION THREAD

**Discussion by skywo1f** [pybamm-team/PyBaMM#2470] [General]
I am running a sensible LFP model based on Prada2013. By sensible, I mean I changed the min and max voltages to 
`       "Lower voltage cut-off [V]": 2.5,
 
       "Upper voltage cut-off [V]": 3.65,
`  
my code is:
`import pybamm

#chemistry = pybamm.parameter_sets.Prada2013
param = pybamm.ParameterValues("Prada2013")

exp = pybamm.Experiment(
    [
        (
            "Charge at 1 C until 3.65 V",
            "Hold at 3.65 V until C/10",
            "Rest for 5 minutes",
            "Discharge at 1 C until 2.5 V",
            "Rest for 5 minutes",
        )
    ]
    )

#model = pybamm.lithium_ion.DFN()
model = pybamm.lithium_ion.SPM()

sim = pybamm.Simulation(model,experiment=exp,parameter_values=param)
sim.solve([0, 3600],initial_soc=0)  # solve for 1 hour
print(sim.parameter_values)

sim.plot(

    [
        "Current [A]",
        "Terminal voltage [V]"
    ]
    )
`

So I start at 0% SoC, charge at 1C, but then the battery is charged in 20 min (see image). Any idea what I am missing?
![lfpBatt](https://user-images.githubusercontent.com/4000822/202541203-34d271ca-989c-4e51-8e34-bd3871eef50d.png)

**Comment by skywo1f** [comment]
the issue seems to have something to do with changing 
`        "Nominal cell capacity [A.h]": 5.0,`
(I changed it from 1.1 to 5). It's like it didn't change the actual capacity of the cell, just changed what amperage C corresponds to.

**Reply by skywo1f** [reply]
(current function and typical current don't seem to change much)

**Reply by skywo1f** [reply]
if I change
`        "Number of electrodes connected in parallel to make a cell": 4.0,
`
(from 1 to 4) then it charges at 1C for an hour. But I am not sure how that makes sense.

**Comment by rtimms** [comment]
`Nominal cell capacity [A.h]` is used to convert between current and C-rate. Since you changed the voltage limits, you changed the useable capacity, so you’ll need to figure out what the new capacity is. 

`Current function [A]` sets the current when you aren’t using an experiment. `Typical current [A]` is just used internally for scaling and shouldn’t  affect the results.

**Comment by skywo1f** [comment]
setting the Ah seems a bit artificial. Is there any way to give Pybamm the battery parameters and have it calculate Ah?

**Reply by skywo1f** [reply]
like shouldnt there be a constant somewhere where It takes the grams of lifepo4 and multiplies by 140 to get total mAh?

**Reply by rtimms** [reply]
Give the electrode capacities, total cyclable lithium, and voltage limits you can calculate the cell capacity as in [this example](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/models/electrode-state-of-health.ipynb). 

The parameter `Nominal cell capacity [A.h]` is simply used to convert between current and C-rate.

**Reply by 1836005678** [reply]
If we change upper and lower voltage limits, then the nominal capacity may also changed.
However, discharge Prada2013 battery at 1C(5A) from default setting(3.6V) to lower voltage cut-off(2.0V) last only 0.75 hour, much shorter than 1 hour. The Prada2013 parameter set may have some problem.

**Reply by rtimms** [reply]
Yeah, I think the Prada set needs some updating (e.g. see #1811). If you take a look at the docstring in [this file](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium_ion/Prada2013.py) you’ll see it’s a combination of data from a number of papers, then fitted to data from the Severson Nature paper from 2019. 

If you make any progress it would be great if you could open a PR to update the parameters.

**Reply by rtimms** [reply]
[This thesis](https://www.diva-portal.org/smash/get/diva2:1593557/FULLTEXT02) might be a helpful starting point.

**Reply by 1836005678** [reply]
After applying your suggestion, simulation result seems correct. Detailed results have been descripted in my comment in #2661.
5 parameters have been updated in PR [https://github.com/pybamm-team/PyBaMM/commit/ac93b8c5cd87683c9f251a787fee886afe4e74b7](url).
Thank you very much!
